base/abstractdict.jl

```diff
@@ -217,6 +217,7 @@ Dict{Int64, Int64} with 3 entries:
 """
 function merge!(d::AbstractDict, others::AbstractDict...)
     for other in others
+        haslength(other) && sizehint!(d, length(d) + length(other))
```

Comment:
Why does d be assumed to haslength, but not other, despite being the same type? Is it because it is mutable?

Reply:
No, just embarrassing. I've added the check in #44687.